thriftbp/client_middlewares.go

```diff
@@ -84,6 +84,11 @@ type DefaultClientMiddlewareArgs struct {
 	//
 	// If it's not set, the global one from ecinterface.Get will be used instead.
 	EdgeContextImpl ecinterface.Interface
+
+	// The value to be set on the User-Agent THeader.
+	//
+	// Optional. If this is empty, no User-Agent header will be sent.
+	UserAgent string
```

Comment:
Should we call this "PeerService" or something, to make the "intent" clear rather than the implementation?

Do we have any way to learn this or set it automatically?

Reply:
Agree on calling it something less tied to the specific header (since we'll use other headers in other protocols) and that auto-detection would be really useful if possible. Python-wise it's almost all auto-detected.

Reply:
there's not a good way to do the auto detection in go. we don't have a single name in [the global config](https://pkg.go.dev/github.com/reddit/baseplate.go#Config) (we do have separated `Metrics.Namespace` and `Tracing.Namespace`), and even if we do, it's not passed here. Having a global variable storing the global name of the current running service and use that for metrics, tracing, and here _might_ make sense but I consider that out of scope for now.

as for the name, the thing on the wire is called `User-Agent`, if we ever use it for things other than `peer.service` calling it `PeerService` might be misleading?

Reply:
how about calling it `ClientName` so it's tied to neither the wire header nor the span tag?